src/bytesearch.rs

```diff
@@ -107,26 +48,16 @@ impl<const N: usize> ByteSeq for [u8; N] {
 impl<const N: usize> ByteSearcher for [u8; N] {
     #[inline(always)]
     fn find_in(&self, rhs: &[u8]) -> Option<usize> {
+        if N == 0 {
+            return Some(0);
+        }
         if N == 1 {
             return memchr::memchr(self[0], rhs);
         }
-        // Use Boyer-Moore for longer patterns to get better performance
-        // than the naive sliding window approach. For very short patterns
-        // (2-3 bytes), the overhead might not be worth it, but for longer
-        // patterns Boyer-Moore provides significant speedup.
-        if N >= 5 {
-            return boyer_moore::boyer_moore_search(self, rhs);
-        }
-
-        // For short patterns (2-3 bytes), still use the simple approach
-        // as the Boyer-Moore overhead might not be worth it
-        for win in rhs.windows(Self::LENGTH) {
-            if self.equals_known_len(win) {
-                // Black magic?
-                return Some((win.as_ptr() as usize) - (rhs.as_ptr() as usize));
-            }
-        }
-        None
+        // Use memchr's highly optimized substring search which uses SIMD when available
+        // and falls back to efficient scalar algorithms like Two-Way when SIMD isn't available.
+        // This is much faster than any custom scalar implementation we could write.
+        memchr::memmem::find(rhs, self)
```

Comment:
If you can find a way to use `memchr::memmem::Finder` so that it is built when the regex is compiled, that would be even better. Otherwise, you're paying to rebuild the `Finder` for every search even though there is an explicit step to pre-compile the regex. i.e., The needle string is invariant.